lib/pipeline.js

```diff
@@ -1147,6 +1149,59 @@ class PipelineModel extends BaseModel {
             .then(() => removeChildPipelines()) // remove pr events
             .then(() => super.remove()); // remove pipeline
     }
+
+    /**
+     * getMetrics for this pipeline
+     * @method getMetrics
+     * @param  {Object}   [config]              Configuration object
+     * @param  {String}   [config.startTime]    Look at events created after this startTime
+     * @param  {String}   [config.endTime]      Look at events created before this endTime
+     * @return {Promise}  Resolves to array of metrics for events belong to this pipeline
+     */
+    async getMetrics(config) {
+        // if no config pass in then get all (will not pass startTime/endTime to datastore)
+        const startTime = config ? config.startTime : null;
+        const endTime = config ? config.endTime : null;
+
+        // Get events during this time range
+        const events = await this.getEvents({
+            startTime,
+            endTime
+        });
+
+        // Calculate event duration by using max endTime - min startTime of builds
+        // that belong this event
+        const eventDuration = async (event) => {
+            const builds = await event.getBuilds();
+
+            // Somehow this event doesn't have any builds
+            if (!builds) {
+                return 0;
+            }
+
+            let minStartTime = new Date(builds[0].startTime);
+            let maxEndTime = new Date(builds[0].endTime);
+
+            for (let i = 1; i < builds.length; i += 1) {
+                const s = new Date(builds[i].startTime);
+                const e = new Date(builds[i].endTime);
+
+                minStartTime = s < minStartTime ? s : minStartTime;
+                maxEndTime = e > maxEndTime ? e : maxEndTime;
+            }
+
+            return Math.round((maxEndTime - minStartTime) / 1000); // round in seconds
+        };
+
+        // Generate metrics
+        const promiseArray = events.map(async (e) => {
```

Comment:
when we pull this in, should probably also increase the connection pool size(current number is 10), this can be a lot of concurrent calls

Reply:
will increase pool size 👍